### PR TITLE
Add an AppVeyor configuration file

### PR DESCRIPTION
**Pull request by sigmavirus24**
Start running some tests on Windows

Related-to: #29

### CODE REVIEW

**Comment by miguelgrinberg** [comment]
A cygwin build of libyaml is really straightforward.

In addition to gcc, which I already had installed (package gcc-g++), I had to install autoconf, automake and libtool. The build did not give me any trouble.

```
./bootstrap
./configure
./make
./make test
./make install
```

Test output:

```
============================================================================
Testsuite summary for yaml 0.1.7
============================================================================
# TOTAL: 2
# PASS:  2
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
```

**Comment by sigmavirus24** [comment]
So then this is just a matter of figuring out how to get those commands to work with AppVeyor :)

**Comment by miguelgrinberg** [comment]
Using cmake I was able to generate a VS project that builds cleanly as well. But unfortunately the cmake configuration does not include anything for tests, it just builds the lib.

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+version: 0.1.7.{build}
+
+image:
+- Visual Studio 2015
+- Visual Studio 2013
+
+
+build_script:
+- cmake --build . --clean-first
```
Comment (miguelgrinberg) [inline appveyor.yml]: This statement runs a build, but as far as I remember it does not generate the project files, makefiles, etc. Above this line, try adding:

    - cmake .

This is what I did to generate the VS project files on my windows box.

**Pull request closed without merging**